src/components/MapView/Analyser/util.ts

```diff
@@ -0,0 +1,18 @@
+export const extractPropsFromURL = (path: string) => {
```

Comment:
Move this into an action along with most the code in your `useEffect` once you start using `useReducer`